File: 7_remove_duplicates/Main.java
public class Main {
    public static void main(String[] args) {
        // Test the method removeDuplicates 
        int[] nums = {1, 1, 2, 2, 3, 4, 4, 5};
        Solution solution = new Solution();
        int newLength = solution.removeDuplicates(nums);
        System.out.println("New length: " + newLength);
        System.out.print("Modified array: ");
        for (int i = 0; i < newLength; i++) {
            System.out.print(nums[i] + " ");
        }
    }
}

class Solution {
    public int removeDuplicates(int[] nums) {
        if (nums == null || nums.length == 0) {
            return 0;
        }
        int unic_index = 1;
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] != nums[i-1]) {
                nums[unic_index] = nums[i];
                unic_index++;
            }
        }
        return unic_index;
    }
}

File: 7_remove_duplicates/main.py
class Solution(object):
    def removeDuplicates(self, nums):
        if not nums:
            return 0
        unic_index = 1
        for i in range(1, len(nums)):
            if nums[i] != nums[i-1]:
                nums[unic_index] = nums[i]
                unic_index += 1
        return unic_index

if __name__ == "__main__":
    solution = Solution()
    nums =[0,0,1,1,1,2,2,3,3,4]
    length = solution.removeDuplicates(nums)
    print("Length of array after removing duplicates:", length)
    print("Array after removing duplicates:", nums[:length])